File: mvc1/_etc/Java/6588 골드바흐의 추측.java
import java.util.*;
import java.io.*;

public class Main {
    
    public static boolean isPrime(int number) {
        if(number == 1) return false;
        if (number % 2 == 0) return false;
        for(int i = 3; i <= (int)Math.pow(number, 0.5); i+=2) {
            if(number % i == 0) return false;
        }
        return true;
    }
    
    public static void main(String args[]) {
        StringBuilder sb = new StringBuilder();
        Scanner sc = new Scanner(System.in);
        
        while(true) {
            int n = sc.nextInt();
            if(n == 0) break;
            
            int j = n;
            for(int i = 1; i < n; i++) {
                j = n-i;
                if(isPrime(i) && isPrime(j)) {
                    sb.append(String.format("%d = %d + %d\n", i+j, i, j));
                    break;
                }
            }
            if(j == 0) sb.append("Goldbach's conjecture is wrong.");
        }
        System.out.println(sb);
    }
}

File: mvc1/_etc/Kotlin/6588 골드바흐의 추측.kt
import java.util.*;
import kotlin.math.*;

fun isPrime(number : Int) : Boolean {
    if(number == 1) return false
    if(number % 2 == 0) return false
    for(i in 3..Math.pow(number.toDouble(), 0.5).toInt()) {
        if(number % i == 0) return false
    }
    return true

}

fun main(args: Array<String>) {
    var sc = Scanner(System.`in`)
    while(true) {
        var n = sc.nextInt();
        if(n == 0) break;
        
        var j = n
        for(i in 1..n-1) {
            j = n-i;
            if(isPrime(i) && isPrime(j)) {
                println("%d = %d + %d".format(i+j, i, j));
                break;
            }
        }
        if(j == 0) println("Goldbach's conjecture is wrong.")
        
    }
    
}
